lightning/src/ln/channelmanager.rs

```diff
@@ -1794,6 +1810,16 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 	}
 
+	/// XXX
+	pub fn keysend(&self, route: &Route) -> Result<PaymentHash, PaymentSendFailure> {
```

Comment:
So have you considered the user providing the preimage ?

E.g a keysend use-case might be sending a preimage as a 32-byte message.

Alternatively you might be interested to return a preimage to the user where the sender is interested to know that the receiver caught the payment, e.g a HTLC-as-a-fidelity-bond kind of thing ?

Reply:
Added user-provided preimages!